# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/shared/CodeGenerator-shared.h
# Commit: 1e582b17eab7
# Full Hash: 1e582b17eab7abf3958e46bc52ea3eeafa764eeb
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 09:30:56
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.6: Add tls dependency for truncate i32. r=lth
#   
#   We generate builtin call for MTruncateToInt32 operation for floating points types,
#   so we need to add a tls dependency.
#   I inserted NYI for arm64 because Ion doesn't support arm64.
# ==============================================================================

diff -r a309060018a8 -r 1e582b17eab7 js/src/jit/shared/CodeGenerator-shared.h
--- a/js/src/jit/shared/CodeGenerator-shared.h	Wed Sep 09 06:37:59 2020 +0000
+++ b/js/src/jit/shared/CodeGenerator-shared.h	Wed Sep 09 08:59:55 2020 +0000
@@ -279,11 +279,16 @@
 
   OutOfLineCode* oolTruncateDouble(
       FloatRegister src, Register dest, MInstruction* mir,
-      wasm::BytecodeOffset callOffset = wasm::BytecodeOffset());
+      wasm::BytecodeOffset callOffset = wasm::BytecodeOffset(),
+      bool preserveTls = false);
   void emitTruncateDouble(FloatRegister src, Register dest,
                           MTruncateToInt32* mir);
+  void emitTruncateDoubleBuiltin(FloatRegister src, Register dest,
+                                 MWasmBuiltinTruncateToInt32* mir);
   void emitTruncateFloat32(FloatRegister src, Register dest,
                            MTruncateToInt32* mir);
+  void emitTruncateFloat32Builtin(FloatRegister src, Register dest,
+                                  MWasmBuiltinTruncateToInt32* mir);
 
   void emitPreBarrier(Register elements, const LAllocation* index);
   void emitPreBarrier(Address address);